### PR TITLE
DOCSP-36027: Shorten class headings

### PR DESCRIPTION
**Pull request by norareidy**
This PR includes all the tickets in this [JIRA epic](https://jira.mongodb.org/browse/DOCSP-36027), which shorten the titles of API documentation pages.

Note: I also restructured the TOC a little by adding new drawers with class methods. This might be slightly out of scope for DOCSP-36027, but it seemed like the best way to reduce page titles enough so that they fit in the sidebar but don't omit necessary context (i.e., which class they belong to).  

Staging: https://preview-mongodbnorareidy.gatsbyjs.io/php-library/DOCSP-36027/reference/

### CODE REVIEW

File: docs/reference/class/MongoDBBulkWriteResult.txt
Comment (GromNaN) [inline docs/reference/class/MongoDBBulkWriteResult.txt:19]: The the double backslashes are usually used for escaping but it looks like it's not necessary. On staging, I see double backslashes in all namespaces. A single backslash should be displayed.

image

File: docs/reference/class/MongoDBModelIndexInfo.txt
Comment (GromNaN) [inline docs/reference/class/MongoDBModelIndexInfo.txt:4]: How can I access this page? When I click on this class name, it opens the list of method but doesn't open this page.

image

File: docs/reference/method/MongoDBClient-addSubscriber.txt
Comment (GromNaN) [inline docs/reference/method/MongoDBClient-addSubscriber.txt:1]: Adding the comment here for the thread. Not related to this file.

Not a regression but a bug that you might fix here: the page on the `MongoDB\Client` class have lost the list of methods
https://www.mongodb.com/docs/php-library/current/reference/class/MongoDBClient/

image

**Review by GromNaN — COMMENTED** [review]
While reviewing the preview page, I spotted some issues that might be related to your change.

> Staging: https://preview-mongodbnorareidy.gatsbyjs.io/php-library/DOCSP-36027/reference/

Why is this particular page empty?

**Review by mongoKart — APPROVED** [review]
shortened headings LGTM, assuming DBX is on board!

File: docs/reference/class/MongoDBBulkWriteResult.txt
Comment (norareidy) [reply docs/reference/class/MongoDBBulkWriteResult.txt:19]: Just to clarify: every namespace mentioned in the docs should use one instead of two backslashes? If so, I'm happy to fix the backslashes in this PR  - or make a new ticket, since it seems like all the namespaces currently have two backslashes and many files would need to be modified

File: docs/reference/class/MongoDBModelIndexInfo.txt
Comment (norareidy) [reply docs/reference/class/MongoDBModelIndexInfo.txt:4]: I just updated the snooty.toml file to make the "Enumeration Classes" drawer clickable (similar to the "Result Classes" and "Write Result Classes" drawers). Now you can access this page by going to "Enumeration Classes" -> scrolling down to "MongoDB\\Model\\IndexInfo"

File: docs/reference/method/MongoDBClient-addSubscriber.txt
Comment (norareidy) [reply docs/reference/method/MongoDBClient-addSubscriber.txt:1]: I actually noticed this about all the "Methods" sections (on MongoDB\Client, MongoDB\Database, MongoDB\Collection, etc). I'll fix the rendering issues in this PR

Comment (norareidy) [reply docs/reference/method/MongoDBClient-addSubscriber.txt:1]: Alright fixed!

File: docs/reference/class/MongoDBBulkWriteResult.txt
Comment (GromNaN) [reply docs/reference/class/MongoDBBulkWriteResult.txt:19]: It's always one backslash. The double backslash doesn't exist in PHP namespaces.

**Comment by norareidy** [comment]
> While reviewing the preview page, I spotted some issues that might be related to your change. Could you fix them directly in this PR, or create jira issues for later?
> 
> > Staging: https://preview-mongodbnorareidy.gatsbyjs.io/php-library/DOCSP-36027/reference/
> 
> Why is this particular page empty?

Hi Jerome, this page isn't actually accessible from the docs site (other than by editing the URL directly) as it's just a drawer for all the API documentation and doesn't have any content. I just added it as the staging link since all the edited pages fall under this heading. Sorry for the confusion!

File: docs/reference/class/MongoDBBulkWriteResult.txt
Comment (jmikola) [reply docs/reference/class/MongoDBBulkWriteResult.txt:19]: The double backslashes were necessary for escaping syntax in RST; however, I think that was only relevant for actual headings and not cases where text is monospace-formatted.

File: docs/reference/class/MongoDBBulkWriteResult.txt
Comment (jmikola) [inline docs/reference/class/MongoDBBulkWriteResult.txt:4]: If all of these pages are nestled under "API Reference", I suggest we just get rid of the " Class" suffix to conserve space in the menus. It's redundant and tends to add extra word wrapping to the menu.

I'll defer to @alcaeus and @GromNaN, though.

File: docs/reference/write-result-classes.txt
Comment (jmikola) [inline docs/reference/write-result-classes.txt:34]: What is the reasoning for eliminating the list of linked methods from `write-result-classes.txt` and `result-classes.txt`? The original point of these pages were to avoid cluttering the navigation.

So instead of a concrete landing page for each result class (like we have with `MongoDBClient.txt`), we consolidated all of the content on these two pages.

I see this PR adds concrete landing pages for the result classes, but the only purpose seems to be including the list of methods. The "Description" headings that describe each class are still _here_, which is inconsistent with other class pages like `MongoDBClient.txt`.

I'd suggest either of the following:

 * Move "Definition" (the only remaining content on these pages) to the individual result class pages, and eliminate `write-result-classes.txt` and `result-classes.txt` altogether. In this case, we can lump all of the classes under a single "Result Classes" heading (no real need to differentiate write results).
 * Move the list of methods back to `write-result-classes.txt` and `result-classes.txt` and eliminate the need for individual pages for these classes (only keeping their methods).

File: docs/reference/method/MongoDBClient__construct.txt
Comment (jmikola) [inline docs/reference/method/MongoDBClient__construct.txt:2]: I assume this was done to consolidate entries in the nav menu, but it means the page title itself is much more vague. Is there no way to preserve `MongoDB\Client::__construct()` here and only trim the navigation entry? One of my original suggestions (perhaps in Slack) was to modify the logic to generate menu content and apply some string manipulation. That would certainly require some tooling changes, but I think it's preferable to leaving the destination pages themselves without identifying titles.

As is, this page is going to show up in search results as "__construct() -- PHP Library Manual", which looks pretty bad.

File: docs/reference/functions.txt
Comment (jmikola) [inline docs/reference/functions.txt:22]: What does `toctree` actually do if it's necessary to duplicate the list of pages in an RST here? Previously, `toctree` was visible in the rendered page output. After the Snooty migration, does it now _only_ control the rendering of the left-hand navigation?

File: docs/reference/class/MongoDBBulkWriteResult.txt
Comment (GromNaN) [reply docs/reference/class/MongoDBBulkWriteResult.txt:4]: I don't think it's a problem since it's on the right. And it's a bit more explicit.

File: docs/reference/class/MongoDBBulkWriteResult.txt
Comment (GromNaN) [reply docs/reference/class/MongoDBBulkWriteResult.txt:19]: Tracked by @norareidy in DOCSP-36627

File: docs/reference/functions.txt
Comment (norareidy) [reply docs/reference/functions.txt:22]: Yes I believe it just affects the organization of the left-hand navigation

File: docs/reference/write-result-classes.txt
Comment (norareidy) [reply docs/reference/write-result-classes.txt]: Okay, fixed according to your first suggestion (moved the "Definition" sections to the individual class pages)

File: docs/reference/class/MongoDBBulkWriteResult.txt
Comment (jmikola) [inline docs/reference/class/MongoDBBulkWriteResult.txt:1]: Are these anchors still needed if the result classes now have their own pages? No need to clean this up now, as it can be addressed in a subsequent PR. It would entail both removing this and updating any links that refer to it (for all such redundant anchors).

File: docs/reference/class/MongoDBChangeStream.txt
Comment (jmikola) [inline docs/reference/class/MongoDBChangeStream.txt:1]: Same comment as `_php-bulk-write-result` above.

File: docs/reference/class/MongoDBModelDatabaseInfoIterator.txt
Comment (jmikola) [inline docs/reference/class/MongoDBModelDatabaseInfoIterator.txt:4]: I can't say what's responsible for this, but I noticed some of the links under "Enumeration Classes" leave the UI in a strange state. For instance, https://preview-mongodbnorareidy.gatsbyjs.io/php-library/DOCSP-36027/reference/class/MongoDBModelDatabaseInfoIterator/ scrolls past all of the main content.

This could be due to the long titles. In this case, I'd be amenable to stripping the full namespace from the class page titles and preserving the fully qualified class name (FQCN) below in the Definition section.

So this can just be "DatabaseInfoIterator Class" if that helps with navigation. I think that would still be fine for SEO purposes.

File: docs/reference/enumeration-classes.txt
Comment (jmikola) [inline docs/reference/enumeration-classes.txt:22]: Since the "Enumeration Classes" drawer is now clickable and expands (per https://github.com/mongodb/mongo-php-library/pull/1229/files#r1484800559), is this content even viewable anymore? I found no way to navigate to this page directly in the current staging deployment.

**Review by jmikola — COMMENTED** [review]
Note: https://github.com/mongodb/mongo-php-library/pull/1229#discussion_r1486444603 is still outstanding.

File: docs/reference/method/MongoDBClient__construct.txt
Comment (jmikola) [reply docs/reference/method/MongoDBClient__construct.txt:2]: I think we need some input from the docs platform team to see what's possible here. If there's not an immediate solution available, I think reducing the page titles to just the base class name and method (e.g. `MongoDB\Client::__construct()` -> `Client::__construct()`) would be the best compromise to allow meaningful page titles.

Comment (norareidy) [reply docs/reference/method/MongoDBClient__construct.txt:2]: I reached out to docs platform and it seems like there is a way to do this! I'll change all the page titles back to the full namespace and just trim the navigation entry

File: docs/reference/class/MongoDBModelDatabaseInfoIterator.txt
Comment (norareidy) [reply docs/reference/class/MongoDBModelDatabaseInfoIterator.txt:4]: Seems like shortening the title fixed the scrolling issue

File: docs/reference/enumeration-classes.txt
Comment (norareidy) [reply docs/reference/enumeration-classes.txt:22]: It's not viewable right now (other than by editing the URL or using the right/left arrow buttons at the bottom of each page). I can make it viewable, or leave as is and remove the content on this page

File: docs/reference/class/MongoDBBulkWriteResult.txt
Comment (norareidy) [reply docs/reference/class/MongoDBBulkWriteResult.txt:1]: If you'd prefer I remove the content from "Enumeration Classes", "Result Classes", etc (mentioned in [this comment](https://github.com/mongodb/mongo-php-library/pull/1229/files#r1488489592)) then yes these anchors aren't necessary and I'll remove them. Otherwise, I'll keep the anchors if I make those landing pages clickable.

File: docs/reference/class/MongoDBModelDatabaseInfoIterator.txt
Comment (jmikola) [reply docs/reference/class/MongoDBModelDatabaseInfoIterator.txt:4]: This does still happen with some other pages: https://preview-mongodbnorareidy.gatsbyjs.io/php-library/DOCSP-36027/reference/method/MongoDBModelCollectionInfo-isCapped/

I imagine this isn't specific to PHPLIB, though, and is probably a more general issue with the docs theme. Perhaps worthy of a DOCSP ticket to be addressed later.

**Comment by jmikola** [comment]
Noted that @schmalliso pointed you to a work-around for preserving the long page titles while still shortening their appearance in the menu (see: [Slack thread](https://mongodb.slack.com/archives/C018UTGP23T/p1707924824236599)).

File: docs/reference/method/MongoDBClient__construct.txt
Comment (jmikola) [reply docs/reference/method/MongoDBClient__construct.txt]: I saw the [Slack thread](https://mongodb.slack.com/archives/C018UTGP23T/p1707924824236599). Glad there's an existing solution for this, and thanks @schmalliso for the assist.

**Comment by jmikola** [comment]
@norareidy: One last request on my end. Can we just eliminate "Enumeration Classes" and merge that with the "Result Classes" heading? I think the differentiation only made sense when there was an intermediary page showing all classes at once. Now that it's just a menu element and each class has its own page, I can't think of any reason to keep them separated.

Hopefully this also makes the docs configuration a bit simpler.

File: docs/reference/class/MongoDBModelDatabaseInfoIterator.txt
Comment (norareidy) [reply docs/reference/class/MongoDBModelDatabaseInfoIterator.txt:2]: Ah, yeah I think this issue has come up before. I'll follow up with Docs platform.

**Review by jmikola — APPROVED** [review]

**Comment by norareidy** [comment]
> @norareidy: Is this ready to merge on your end? If so, I can do so manually since the current repository rules are preventing you from doing so (see: #1231).

Yes it is! I was waiting for the checks to finish, but if you can bypass them that would be great.

**Pull request merged**
